File: api/app/graphql/mutations/recordings/add_like_to_recording.rb
module Mutations::Recordings
  class AddLikeToRecording < Mutations::BaseMutation
    field :errors, [String], null: false
    field :like, Types::LikeType, null: true
    field :success, Boolean, null: false

    argument :recording_id, ID, required: true

    def resolve(recording_id:)
      check_authentication!

      like = Like.new(likeable_type: "Recording", likeable_id: recording_id, user: current_user)
      if like.save
        {like:, success: true, errors: []}
      else
        {like: nil, success: false, errors: like.errors.full_messages}
      end
    end
  end
end

File: native/graphql/mutations/add_like_to_recording.ts
import { gql } from "@apollo/client";

export const ADD_LIKE_TO_RECORDING = gql`
  mutation AddLikeToRecording($recordingId: ID!) {
    addLikeToRecording(input: { recordingId: $recordingId }) {
      like {
        id
        likeableType
        likeableId
        user {
          id
        }
      }
      success
      errors
    }
  }
`;

export default ADD_LIKE_TO_RECORDING;
